### PR TITLE
Reverted back to using EndpointName in the context of routing.

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Reverted back to using `EndpointName` instead of `Endpoint` in the context of routing. `Endpoint` is back in the root and is used only as a static factory.

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
Please review.

File: src/NServiceBus.Core/Routing/EndpointName.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        /// <summary>
+        /// Tests for inequality.
+        /// </summary>
+        public static bool operator !=(EndpointName left, EndpointName right)
+        {
+            return !Equals(left, right);
+        }
+
+    }
+}
```
Comment (timbussmann) [inline src/NServiceBus.Core/Routing/EndpointName.cs]: run a code cleanup on that file please (and remove the readonly)

Comment (SzymonPobiega) [inline src/NServiceBus.Core/Routing/EndpointName.cs]: :+1: done.

**Comment by SzymonPobiega** [comment]
@timbussmann can you hit the nice green button when ATs pass?

**Comment by timbussmann** [comment]
done.

**Comment by andreasohlund** [comment]
I realized that we need to updates the downstream transports to use this new api (was trying to do a pull for #3242 for sql and got stuck on

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0534	'SqlServerTransport' does not implement inherited abstract member 'TransportDefinition.BindToLocalEndpoint(EndpointInstance, ReadOnlySettings)'	NServiceBus.SqlServer	C:\dev\NServiceBus.SqlServer\src\NServiceBus.SqlServer\SqlServerTransport.cs	16	Active

```

**Pull request merged**